pkg/local_object_storage/writecache/persist.go

```diff
@@ -79,7 +92,33 @@ func (c *cache) persistToCache(objs []objectInfo) []int {
 
 	var failDisk []int
 
+	cacheSz := c.estimateCacheSize()
+
 	for _, i := range failMem {
+		var (
+			updCacheSz  uint64
+			overflowInd = -1
+		)
+
+		if failMem[i] < 0 {
+			// actually, since the overflow was detected in DB tx, the required space could well have been freed,
+			// but it is easier to consider the entire method atomic
+			overflowInd = -failMem[i]
+		} else {
+			// check if object will overflow write-cache size limit
+			if updCacheSz = c.incSizeFS(cacheSz); updCacheSz > c.maxCacheSize {
+				overflowInd = i
```

Comment:
It looks a bit suspicious when `overflowInd` sets the index of `failMem` slice there, but it used as the index of `objs` slice down there. Is it correct?

Reply:
It is correct here: `i` is not an index of `failMem` slice but the value. But the remark helped to find the error above - in `if` block. Renamed to `objInd` for readability and fixed.